strawberryfields/apps/sample.py

```diff
@@ -361,3 +377,66 @@ def vibronic(
     if np.any(t == 0):
         s = np.pad(s, ((0, 0), (0, n_modes))).tolist()
     return s
+
+
+def waw_matrix(A: np.ndarray, w: Union[np.ndarray, list]) -> np.ndarray:
+    r"""Rescale adjacency matrix to account for node weights.
+
+    Given a graph with adjacency matrix :math:`A` and a vector :math:`w` of weighted nodes,
+    this function rescales the adjacency matrix according to:
+
+    .. math::
+
+        A \rightarrow WAW,
+
+    with :math:`W` the diagonal matrix formed by the weighted nodes. This encoding can be used to
+    add a bias in GBS toward selecting subgraphs with a large product of node weights
+    :cite:`banchi2019molecular`. The resulting matrix can be passed to :func:`sample`.
+
+    **Example usage:**
+
+    >>> g = nx.erdos_renyi_graph(5, 0.7)
+    >>> a = nx.to_numpy_array(g)
+    >>> w = [1, 1, 3, 1, 0.5]
+    >>> a = waw_matrix(a, w)
+    >>> sample(a, 3, 4)
+    [[0, 0, 0, 0, 0], [1, 0, 0, 0, 1], [0, 0, 0, 0, 0], [0, 1, 1, 1, 1]]
```

Comment:
Can we `post-select` on samples that more clearly show that nodes with large weight appear more often?

Reply:
```rst
>>> g = nx.erdos_renyi_graph(5, 0.7)
>>> a = nx.to_numpy_array(g)
>>> w = [10, 1, 0, 1, 1]
>>> a = waw_matrix(a, w)
>>> sample(a, 3, 4)
[[1, 1, 0, 1, 0], [1, 1, 0, 1, 0], [1, 1, 0, 0, 0], [1, 1, 0, 1, 0]]
```
This is what I came up with. There is a zero weight one which nicely never clicks. There is a 10 weight one which always clicks

Reply:
Or maybe use large weights such that all samples reflect that without post selecting?

Reply:
Thanks, yes we're now using some more biased weights and picking some indicative samples.